syntax = "proto3";

package secretsharing;

// Defines the service
service SecretSharing {
    // Function invoked to send the request
    rpc GetAttestation (AttestationRequest) returns (AttestationReply) {}
    rpc VerifyAttestation (AttestationRequest) returns (AttestationReply) {}
    rpc DecryptAttachedData (EncryptedData) returns (Reply) {}
}

enum CommandRequest {
    ATTESTATION = 0;
    SYNC_SYM = 1;
}

// The request message containing requested numbers
message AttestationReply {
    bool ok = 1;
    int32 key_kind = 2;
    repeated int32 report = 3;
    repeated int32 key = 4;
    string msg = 5;
}

// The request message containing requested numbers
message AttestationRequest {
    CommandRequest cmd = 1;
    repeated int32 report = 2;
    repeated int32 key = 3;
}

message EncryptedData {
    repeated int32 data = 1;
}

message Reply {
    bool ok = 1;
    string msg = 2;
}
